src/spl_core/test_utils/spl_build.py

```diff
@@ -75,3 +103,66 @@ def execute(self, target: str, additional_args: Optional[List[str]] = None) -> i
             else:
                 break
         return return_code
+
+    def create_artifacts_archive(self, expected_artifacts: List[Path]) -> Path:
+        """
+        Create a zip file containing the collected artifacts.
+
+        Args:
+            expected_artifacts: List of Path of artifacts which should be archived
+
+        Returns:
+            Path: The path to the created zip file.
+
+        Raises:
+            Exception: If there is an error creating the zip file.
+
+        """
+        zip_path = self.build_dir / "artifacts.zip"
+
+        # Delete the file if it already exists
+        if zip_path.exists():
+            zip_path.unlink()
+
+        try:
+            with zipfile.ZipFile(zip_path, "w") as zip_file:
+                artifacts_collection = ArtifactsCollection(artifacts=expected_artifacts, build_dir=self.build_dir)
+                for artifact in artifacts_collection.archive_artifacts:
+                    zip_file.write(artifact.absolute_path, arcname=artifact.archive_path)
+            print(f"Zip file created at: {zip_path}")
+            return zip_path
+        except Exception as e:
+            print(f"Error creating artifacts zip file: {e}")
+            raise e
+
+    def create_artifacts_json(self, expected_artifacts: List[Path]) -> Path:
+        """
+        Create a JSON file listing the collected artifacts.
+
+        Returns:
+            Path: The path to the created JSON file.
+
+        Raises:
+            Exception: If there is an error creating the JSON file.
+
+        """
+        artifacts_collection = ArtifactsCollection(artifacts=expected_artifacts, build_dir=self.build_dir)
+        json_content = {
+            "variant": self.variant,
+            "build_kit": self.build_kit,
+            "artifacts": [str(artifact.archive_path.as_posix()) for artifact in artifacts_collection.archive_artifacts],
+        }
+        json_path = self.build_dir / "artifacts.json"
+
+        # Delete the file if it already exists
```

Comment:
Simplify the code to only write the json content: `json_path.write_text(json.dumps(json_content, indent=4))`